### PR TITLE
GP migration updates for July 2023

### PR DESCRIPTION
**Pull request by jaymckinney-enavate**
This PR includes July 2023 updates to the GP migration.

**General Performance**
- Don't perform the staging combine step if the module is not configured to migrate. An initial step for a migration is the combining of multiple staging tables into a single table for later processing (replacement of the SQL queries). This area of the code is now optimized to only perform this operation when configured to and skip otherwise. If configured to migrate less data, this would reduce both the time a migration would take, and the amount of space used by the staging tables.

**Posting enhancements**
- Allow skip posting. Added 'skip posting' config settings and incorporated them in the posting code to give the ability to skip auto posting of Account, Customer, Bank, and/or Vendor batches.
- Posting Group account number now defaults to Posting Setup account when the Posting Group account number is empty.

**AR/AP**
- Prevent $0 transactions that cause posting to fail.
- Enhancements to Vendor Bank Account migrations. If the Vendor has more than 1 bank account, then the Code field will be appended with a number after the Vendor ID. For example: ACETRAVEL-1, ACETRAVEL-2. All Bank Account records migrated will set the Use for Electronic Payments field = true.
- Remove appended BC doc number to External Document No. field in AP and AR documents. This area of the code was appending the Document No. that is assigned to the migrated AP and AR transactions to the end of the External Doc. No. field. Feedback has been provided by Partners and Customers that this causes confusion, is unnecessary and can break integrations.
- Correct how Countries/Regions are created during migration. The migration now has the ability to perform a search on known countries, obtained by GP's list of known countries.

**Item**
- Support GP Rounding Precision For Item Unit Cost. This change updates the General Ledger Setup record "Unit-Amount Rounding Precision" value to match the greatest number of decimal places supported by a GP Item.

**Historical Snapshot improvements**
- Restart GP Snapshot if it times out (Increased the Max Attempts property).
- Snapshot performance enhancements.

**Other**
- OnResetAllCloudData() and CompanyOnAfterDelete() now cleans up necessary tables.
- Ablity to update Company migration settings and rerun the Snapshot with the changed settings.
- Added a page to add the ability to remove a Hybrid Company Status record

### CODE REVIEW

File: Apps/W1/HybridGP/app/src/pages/GPCompanyMigrationSettings.Page.al [hunk 1]
```diff
@@ -0,0 +1,155 @@
+page 40056 "GP Company Migration Settings"
```
Comment (nikolakukrika) [inline Apps/W1/HybridGP/app/src/pages/GPCompanyMigrationSettings.Page.al:1]: I believe this object should be added to the permission sets, together with the new query

File: Apps/W1/HybridGP/app/src/pages/GPCompanyMigrationSettings.Page.al [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+{
+    ApplicationArea = All;
+    Caption = 'GP Company Migration Settings';
+    PageType = List;
+    SourceTable = "GP Company Additional Settings";
+    SourceTableView = sorting(Name) where("Name" = filter(<> ''));
+    UsageCategory = Lists;
```
Comment (nikolakukrika) [inline Apps/W1/HybridGP/app/src/pages/GPCompanyMigrationSettings.Page.al:8]: Minor - Worksheet is usually good for this kind of tables that should always be editable.

**Review by nikolakukrika — DISMISSED** [review]

File: Apps/W1/HybridGP/app/src/pages/GPCompanyMigrationSettings.Page.al (hunk 1 above)
Comment (jaymckinney-enavate) [reply Apps/W1/HybridGP/app/src/pages/GPCompanyMigrationSettings.Page.al:1]: Updated, thanks

File: Apps/W1/HybridGP/app/src/pages/GPCompanyMigrationSettings.Page.al (hunk 2 above)
Comment (jaymckinney-enavate) [reply Apps/W1/HybridGP/app/src/pages/GPCompanyMigrationSettings.Page.al:8]: Updated, thanks

**Review by nikolakukrika — APPROVED** [review]

**Comment by JesperSchulz** [comment]
Thanks for reporting this. We agree, and we’ll publish a fix asap, either in an update for the current version or in the next major release. Please do not reply to this, as we do not monitor closed issues. If you have follow-up questions or requests, please create a new issue where you reference this one.

Build ID: 11735.

**Comment by JesperSchulz** [comment]
Availability update: We will publish a fix for this issue in the next update for release 22.

Build ID to track: 59590.

**Pull request closed without merging**